definitions/procedures/backup/metadata.rb

```diff
@@ -0,0 +1,44 @@
+module Procedures::Backup
+  class Metadata < ForemanMaintain::Procedure
+    metadata do
+      description 'Generate metadata'
+      tags :backup
+      param :backup_dir, 'Directory where to backup to', :required => true
+    end
+
+    def run
+      with_spinner('Collecting metadata') do |spinner|
+        spinner.update('Collecting system release info')
+        os_version = execute!("cat /etc/redhat-release").chomp
+
+        spinner.update('Collecting list of plugins')
+        plugins = plugin_list
+
+        spinner.update('Collecting installed RPMs')
+        rpms = execute!("rpm -qa").split("\n")
+
+        system_facts = {
+          os_version: os_version,
+          plugin_list: plugins,
```

Comment:
I think it would make sense to have `plugin_list` and `proxy_features_list` separately